docs: document the fiocli updates upload feature (#217)
foundriesio/dg-satellite: docs/updates.md

## Producing and Serving Update Content

Update content is the *exact* contents of what `fioctl targets
offline-update` produces (an `ostree_repo` and/or an `apps` directory,
and optionally a `tuf` directory).

> ### Uploading an update

Reviewer: ```suggestion
### Uploading a Update
```

The recommended way to publish update content is the

> `fiocli updates upload` command, which streams a local offline-update

Reviewer: ```suggestion
The recommended way to publish update content is
`fiocli updates upload`, which streams a local offline-update
```

directory to the server over the API:

```
  fiocli updates upload <tag> <update-name> <directory>
```

For example, to upload the offline content produced above and name the
update `148` under the `main` tag:

```
  fioctl targets offline-update --expires-in-days 180 --tag main intel-corei7-64-lmp-148 ./148
  # If you've enabled TUF on your server, then you should remove the `tuf` 
  # directory before running fiocli updates upload
  # rm -rf ./148/tuf
  fiocli updates upload main 148 ./148
```

The directory is archived, gzip-compressed, and streamed to the server.
This update will then show up under the "updates" view in your UI.

> ### Automatic TUF target generation

Reviewer: ```suggestion
### Automatic TUF Target Generation
```

directory, the server generates a TUF target for the update itself
(this requires TUF to be enabled on the server).

To build the target, the server *probes* the uploaded content to make a
best-effort guess about the target's attributes:

* **`ostree-hash`** and **`name`** are read from the ostree repository's
  ref under `ostree_repo`.
* **`version`** (AppVersion) is read from `IMAGE_VERSION` in the image's
  `/usr/lib/os-release`.
* **`hardware-id`** is read from `primary_ecu_hardware_id` in the image's
  `/usr/lib/sota/conf.d/40-hardware-id.toml`. If that file is absent, the
  server falls back to detecting the architecture from the image
  (`arm64-linux` or `amd64-linux`).
* **`apps`** are discovered from the `apps` directory layout, mapping
  each app to its sha256.

Because probing is a best effort, you can override any of these attributes
with flags, which is also useful when uploading content that has no
`ostree_repo` (for example, an apps-only update):

```
  fiocli updates upload main 148 ./148 \
    --hardware-id intel-corei7-64 \
    --version 148 \
    --name intel-corei7-64-lmp-148 \
    --ostree-hash <sha256> \
    --apps shellhttpd=<sha256>
```

> > **Note:** If you are bundling apps into your updates, probing the

Reviewer: ```suggestion
> [!NOTE]
> If you are bundling apps into your updates, probing the
```

> > ostree image for a default `version` won't be sufficient. Your apps

Reviewer: ```suggestion
> ostree image for a default `version` will not be sufficient. Your apps
```

> might change from one update to the next while still using the same
> ostree, so the probed `IMAGE_VERSION` would collide. In that case,
> supply a distinct `--version` for each update.

> > **Note:** When the upload does not contain an `ostree_repo` directory,

Reviewer: ```suggestion
> [!NOTE]
> When the upload does not contain an `ostree_repo` directory,
```
